### DISCUSSION TITLE
IDH Playbook Alerts Not Functioning

### DISCUSSION THREAD

**Discussion by jtrodriguez** [Security-Onion-Solutions/securityonion#8174] [Unsupported Versions]
I am no longer seeing IDH Playbooks alerts show up on my Alerts page and my custom e-mail alerts for IDH are no longer functioning.

Did something change in a recent update with how IDH is logged?

**Comment by defensivedepth** [comment]
Nothing has changed with how IDH is logged. What SO version are you on?

A few things to check:

- Is the IDH service actually running (on the IDH node, check with `sudo so-status`

- Do you see IDH logs in Hunt or Kibana?

- Are there `enabled` IDH plays in Playbook?

**Reply by jtrodriguez** [reply]
Version 2.3.130
I see them in both Hunt and Kibana. Just not under alerts.

**Reply by jtrodriguez** [reply]
and yes, they are all active in playbook and the service is running on the node.

**Reply by jtrodriguez** [reply]
I ran so-playbook-reset in hopes of it fixing the issue since it seems to be playbook related. I am now missing the IDH plays because they don't seem to belong to the default repository.

**Reply by defensivedepth** [reply]
If you have IDH logs in Hunt & Kibana, can you confirm that you have logs that would generate an alert?

To re-import the IDH plays, run `so-playbook-import` - you may have to enable that after the import.

**Reply by jtrodriguez** [reply]
Thank you for the tip on so-playbook-import. I got them back and enabled them again.

I am intentionally triggering the IDH with a login and it shows up in Kibana when I search for "opencanary" the entry has my IP and password that I attempted to use among all the other info.

**Reply by jtrodriguez** [reply]
I just noticed that I am getting playbook alerts for playbook plays that are not active. Somehow the playbook plays are not syncing to the alerts or are cached. I am not sure where to go from here.

**Reply by defensivedepth** [reply]
@jtrodriguez Getting back from vacation and catching up on things :)

Delete all `*.yaml` files in `/opt/so/rules/elastalert/playbook/` then run `sudo so-playbook-sync` - that should regenerate the elastalert rules for all active Plays.

On one of the opencanary logs that should be triggering an alert - what is the value of the `event.code` field? The IDH Plays generate  alerts off of the event.code fields on OpenCanary logs.

**Reply by jtrodriguez** [reply]
Thank you for assisting. I was on phone support for over an hour attempting to resolve this issue. I have another call scheduled for tomorrow. 
The alerts started working then stopped working then started working again and now today I have no alerts at all for the last 24 hours.  I am hoping to resolve this tomorrow on the support call and maybe summarize the solution here if it's simple enough.
